homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        update_interval: int,
+        ip_address: str,
+        scan_coordinator: FluxLEDListUpdateCoordinator = None,
+    ):
+        """Initialize the update coordinator."""
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(milliseconds=update_interval),
+        )
 
-        self._bulb = WifiLedBulb(self._ipaddr, timeout=5)
-        if self._protocol:
-            self._bulb.setProtocol(self._protocol)
+        self._ip_address = ip_address
+        self._name = name
+        self.scan_coordinator = scan_coordinator
+
+        self.light = WifiLedBulb(self._ip_address)
+
+    async def _async_update_data(self):
+        """Fetch the data from this light bulb."""
+
+        await self.hass.async_add_executor_job(self.light.update_state)
+
+        get_rgbw = await self.hass.async_add_executor_job(self.light.getRgbw)
+        get_rgb = await self.hass.async_add_executor_job(self.light.getRgb)
+
+        light = {}
+
+        if self.light.mode == "ww":
+            light["mode"] = MODE_WHITE
+        elif self.light.rgbwcapable:
+            light["mode"] = MODE_RGBW
+        else:
+            light["mode"] = MODE_RGB
+
+        light["white_value"] = get_rgbw[3]
+
+        if light["mode"] == MODE_WHITE:
+            light["brightness"] = light["white_value"]
+        else:
+            light["brightness"] = self.light.brightness
+
+        light["hs_color"] = color_util.color_RGB_to_hs(*get_rgb)
+
+        light["current_effect"] = self.light.raw_state[3]
+
+        if self.light.is_on and light["brightness"] > 0:
+            light["state"] = True
+        else:
+            light["state"] = False
+
+        return light
+
+
+class FluxLight(CoordinatorEntity, LightEntity):
+    """Represents a Flux Light entity."""
+
+    def __init__(
+        self,
+        coordinator: FluxLEDCoordinator,
+        unique_id: str,
+        device: dict,
+        effect_speed: int,
+        hass: HomeAssistant,
+    ):
+        """Initialize the Flux light entity."""
+        super().__init__(coordinator=coordinator)
+
+        self._name = device[CONF_NAME]
+        self._unique_id = unique_id
+        self._icon = "mdi:lightbulb"
+        self._attrs = {}
+        self._last_update = 0
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._bulb = coordinator.light
+        self._last_brightness = 255
+        self._last_hs_color = color_util.color_RGB_to_hs(255, 255, 255)
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = coordinator.data["mode"]
+
+        async def async_remove_light(device: dict):
+            """Remove a bulb device when it is removed from options."""
+
+            bulb_id = device["device_id"]
+
+            if self._unique_id != bulb_id:
```

Comment:
If we encode the bulb id in the remove signal we don't need this check here.

Reply:
I'm not sure I can do that (but happy to be wrong). In the config_flow we store the device properties in the config_entry, but the actual entities and ids aren't stored there. So when the user selects an LED to remove in options flow I can pass in the device properties to the call, but not the specific entity which is why I added the check here.

Reply:
Ok, yeah we need to know the bulb id to use in the signal.